internal/controller/controller_test.go

```diff
@@ -596,6 +600,381 @@ var _ = Describe("Testing Route to HTTPProxy Controller", func() {
+					},
+					WildcardPolicy: routev1.WildcardPolicyType(RouteFooWildCardPolicyType),
+				},
+			}
+			err = k8sClient.Create(context.Background(), &objRouteFoo)
+			Expect(err).To(BeNil())
+
+			objRouteFoo.Status = routev1.RouteStatus{
+				Ingress: []routev1.RouteIngress{
+					{
+						RouterName: RouterName,
+						Conditions: []routev1.RouteIngressCondition{
+							{
+								Status: v12.ConditionStatus(v1.ConditionTrue),
+								Type:   routev1.RouteAdmitted,
+							},
+						},
+					},
+				},
+			}
+			err = k8sClient.Status().Update(context.Background(), &objRouteFoo)
+			Expect(err).To(BeNil())
+
+			rObjFoo := routev1.Route{}
+			err = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: DefaultNamespace, Name: RouteFooName}, &rObjFoo)
+			Expect(err).To(BeNil())
+
+			time.Sleep(1 * time.Second)
+			httpProxyList := contourv1.HTTPProxyList{}
+			err = k8sClient.List(context.Background(), &httpProxyList, client.InNamespace(DefaultNamespace))
+			Expect(err).To(BeNil())
+			Expect(len(httpProxyList.Items)).To(Equal(1))
+			Expect(httpProxyList.Items[0].Spec.VirtualHost.Fqdn).To(Equal(RouteFooFQDN))
+
+			rObjFoo = routev1.Route{}
+			err = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: DefaultNamespace, Name: RouteFooName}, &rObjFoo)
+			Expect(err).To(BeNil())
+			rObjFoo.Spec.Host = RouteFooNewFQDN
+			err = k8sClient.Update(context.Background(), &rObjFoo)
+			Expect(err).To(BeNil())
+			rObjFoo = routev1.Route{}
+			err = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: DefaultNamespace, Name: RouteFooName}, &rObjFoo)
+			Expect(err).To(BeNil())
+			rObjFoo.Status = routev1.RouteStatus{
+				Ingress: []routev1.RouteIngress{
+					{
+						RouterName: RouterName,
+						Conditions: []routev1.RouteIngressCondition{
+							{
+								Status: v12.ConditionStatus(v1.ConditionTrue),
+								Type:   routev1.RouteAdmitted,
+							},
+						},
+					},
+				},
+			}
+			err = k8sClient.Status().Update(context.Background(), &rObjFoo)
+			Expect(err).To(BeNil())
+
+			// wait for new HTTPProxy creation, and deletion of old one
+			time.Sleep(2 * time.Second)
+
+			httpProxyList = contourv1.HTTPProxyList{}
+			err = k8sClient.List(context.Background(), &httpProxyList, client.InNamespace(DefaultNamespace))
+			Expect(err).To(BeNil())
+			Expect(len(httpProxyList.Items)).To(Equal(1))
+			Expect(httpProxyList.Items[0].Spec.VirtualHost.Fqdn).To(Equal(RouteFooNewFQDN))
+
+			err = k8sClient.Delete(context.Background(), &objRouteFoo)
+			Expect(err).To(BeNil())
+
+			httpProxyList = contourv1.HTTPProxyList{}
+			err = k8sClient.List(context.Background(), &httpProxyList, client.InNamespace(DefaultNamespace))
+			Expect(err).To(BeNil())
+			Expect(len(httpProxyList.Items)).To(Equal(1))
+		})
+
+		It("should create new HTTPProxy object when there are two routers and the host of older route is changed, the older HTTPProxy should change the controller reference", func() {
```

Comment:
1. router --> route
2. Not sure how to, but I think the message could be improved. For example:
> When the host of one of two same routes changes, should create a new HTTPProxy and change the controller reference of the existing one